.tekton/scaffolding-unit-test.yaml

```
@@ -0,0 +1,24 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: go-unit-test
+  annotations:
+    tekton.dev/title: "Go Unit Test Task"
+spec:
+  workspaces:
+    - name: source
+  steps:
+    - name: run-tests
+      image: registry.access.redhat.com/ubi9/go-toolset@sha256:c3a9c5c7fb226f6efcec2424dd30c38f652156040b490c9eca5ac5b61d8dc3ca
+      workingDir: $(workspaces.source.path)/source
+      script: |
+        #!/usr/bin/env sh
+        cp go.mod go.mod.bak
+        sed -i '/toolchain/d' go.mod
+        go test ./...
```

Instead of copying and restoring, would this work?
```suggestion
        GOTOOLCHAIN=go1.20.0 go test ./...
```
See https://go.dev/doc/toolchain

---

Oh that's funny. The `toolchain` directive was introduced in 1.21 I think. So even though setting `GOTOOLCHAIN` should work to enforce a specific version of Go, when you force 1.20 on a go.mod file with the `toolchain` directive you're still out of luck, because 1.20 can't parse it.